app/src/main/java/org/covidwatch/android/ui/exposures/ExposuresViewModel.kt

```diff
@@ -0,0 +1,59 @@
+package org.covidwatch.android.ui.exposures
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.launch
+import org.covidwatch.android.data.CovidExposureInformation
+import org.covidwatch.android.data.exposureinformation.ExposureInformationRepository
+import org.covidwatch.android.exposurenotification.ENStatus
+import org.covidwatch.android.exposurenotification.ExposureNotificationManager
+import org.covidwatch.android.functional.Either
+
+class ExposuresViewModel(
+    private val enManager: ExposureNotificationManager,
+    exposureInformationRepository: ExposureInformationRepository
+) : ViewModel() {
+
+    private val _exposureNotificationEnabled = MutableLiveData<Boolean>()
+    val exposureNotificationEnabled: LiveData<Boolean> = _exposureNotificationEnabled
+
+    val exposureInfo: LiveData<List<CovidExposureInformation>> =
+        exposureInformationRepository.exposureInformation()
+
+    fun toggleExposureNotifications() {
+        viewModelScope.launch {
+            // Check if we need to stop the service
+            if (enManager.isEnabled().result() == true) {
+                enManager.stop()
+            } else { // Otherwise run the service
+                enManager.start().result()
+            }
+
+            // Handle the case of is service running or not simply by asking the manager
```

Comment:
Why not just directly pipe `enManager.isEnabled()` to `_exposureNotificationEnabled`? Ie a direct observer on the manager. Not necessary in this PR, tho, jw

Reply:
`enManager.isEnabled()` is not an observable type so it fires you update once you ask for it
but you reminded me that I need to update the state of `_exposureNotificationEnabled` on opening this screen